lib/std/heap.zig

```diff
@@ -719,6 +719,54 @@ test "Thread safe FixedBufferAllocator" {
     try testAllocatorAlignedShrink(fixed_buffer_allocator.threadSafeAllocator());
 }
 
+test "FixedBufferAllocator at comptime" {
+    comptime {
+        var buf: [16]u8 = undefined;
+        var fba = FixedBufferAllocator.init(&buf);
+
+        {
+            var slice = fba.allocator().alloc(u8, 8) catch unreachable;
+            slice[3] = 2;
+            assert(slice[3] == 2);
```

Comment:
is it idiomatic to `assert()` in a test that is designed to be in comptime? i could also wrap all the logic in a function and `comptime foo() catch unreachable`. that's more typing, but makes the test logic look more like a test?

Reply:
Yes, that's a reasonable thing to do, because the alternatives are worse:
- `comptime try expect(...)` gives a useless error if the test fails
- `try comptime expect(...)` reports the failure at runtime even though it's known at comptime
- `comptime expect(...) catch unreachable` is very noisy for the same effect as `comptime assert(...)`
- `comptime doAllTheTests() catch unreachable` is a bit noise and gives a worse error

I've been using `comptime assert` whenever I want to verify something at compile-time in behavior tests for a while, and it seems generally better than other solutions.